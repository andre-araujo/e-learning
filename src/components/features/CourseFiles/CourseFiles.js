import React, { Component, Fragment } from 'react';
import { string } from 'prop-types';
import moment from 'moment';

import Hero from '../../modules/Hero';
import {
  FileInput, Text, Button, Wrapper, InternalLink,
} from '../../elements';
import { ModalProvider } from '../../modules';


import { deepSelect } from '../../../lib/utils';

class CourseFiles extends Component {
  state = {
    file: null,
  }

  onSubmit = (e) => {
    e.preventDefault();
    const {
      courseId,
      fileUpload,
    } = this.props;
    const {
      file,
    } = this.state;

    fileUpload({
      courseId,
      file,
    });
  }

  render() {
    const {
      isAdmin,
      courseDetail,
    } = this.props;
    const {
      file,
    } = this.state;

    const courseFiles = deepSelect(courseDetail, 'payload.files', []);

    return (
      <div>
        <Text.Subtitle tag="h1" margin="0 0 10px">
          Arquivos:
        </Text.Subtitle>

        <hr />

        {
          isAdmin && (
            <form onSubmit={this.onSubmit}>
              <FileInput name="coursemedia" onChange={inputFile => this.setState({ file: inputFile })} />
              <Button full disabled={!file}>
                Adicionar Mídia
              </Button>
            </form>
          )
        }

        <ol>
          {
            courseFiles.map((courseFile, index) => (
              <li key={courseFile._id}>
                <a href={courseFile.url} target="_blank" rel="noopener noreferrer">
                  <Text.Small margin="20px 0" style={{ display: 'block' }}>
                    {`${`${index}`} - ${courseFile.name}`}
                  </Text.Small>
                </a>

              </li>
            ))
          }
          {
            !courseFiles.length && (
              <li>
                <Text.Small margin="20px 0" style={{ display: 'block' }}>
                  Ainda não existem arquivos para este curso
                </Text.Small>
              </li>
            )
          }
        </ol>
      </div>
    );
  }
}

export default CourseFiles;
