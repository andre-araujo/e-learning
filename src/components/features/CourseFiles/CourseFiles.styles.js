import styled from 'styled-components';

export const FileItem = styled.li`
  display: flex;
  align-items: center;
  padding: 10px 0;

  button {
    height: 20px;
    width: 20px;
    margin-right: 10px;

    :hover {
      color: orange;
    }
  }
`;

export const Container = styled.div`
  margin-bottom: 10px;
`;
