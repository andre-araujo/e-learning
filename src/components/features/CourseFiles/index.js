import { connect } from 'react-redux';

import CourseFiles from './CourseFiles';
import { actions } from '../../../lib';

const mapStateToProps = ({ courseDetail }) => ({
  courseDetail,
});

const mapDispatchToProps = dispatch => ({
  fileDelete: data => dispatch(actions.fileDelete(data)),
  fileUpload: data => dispatch(actions.fileUpload(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(CourseFiles);
