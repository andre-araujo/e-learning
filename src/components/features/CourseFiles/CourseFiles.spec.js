import React from 'react';
import { mount } from 'enzyme';

import CourseFiles from './CourseFiles';

describe('CourseFiles component', () => {
  it('should match snapshot', () => {
    expect(mount(<CourseFiles />)).toMatchSnapshot();
  });
});
