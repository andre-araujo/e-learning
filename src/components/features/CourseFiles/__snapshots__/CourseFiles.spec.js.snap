// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseFiles component should match snapshot 1`] = `
<div>
  <Component
    margin="0 0 10px"
    tag="h1"
  >
    Arquivos:
  </Component>
  <hr />
  <ol>
    <li>
      <Component
        margin="20px 0"
        style={
          Object {
            "display": "block",
          }
        }
      >
        Ainda n√£o existem arquivos para este curso
      </Component>
    </li>
  </ol>
</div>
`;
